hca2scea-backend/script.py

```diff
@@ -468,38 +527,92 @@ def extract_csv_from_spreadsheet(work_dir, excel_file):
     )
     parser.add_argument(
         "-c",
         "--curators",
         nargs='+',
+        required=True,
         help="space separated names of curators"
     )
+    parser.add_argument(
+        "-t",
+        "--experiment_type",
+        type=str,
+        required=True,
+        choices=['baseline','differential'],
+        help="Please indicate whether this is a baseline or differential experimental design"
+    )
+    parser.add_argument(
+        "-f",
+        "--experimental_factors",
+        nargs='+',
+        required=True,
+        help="space separated list of experimental factors"
+    )
+    parser.add_argument(
+        "-pd",
+        "--public_release_date",
+        type=str,
+        required=True,
+        help="Please enter the public release date in this format: YYYY-MM-DD"
+    )
+    parser.add_argument(
+        "-hd",
+        "--hca_update_date",
+        type=str,
+        required=True,
+        help="Please enter the last time the HCA prohect submission was updated in this format: YYYY-MM-DD"
+    )
+    parser.add_argument(
+        "-r",
+        "--related_scea_accession",
+        nargs='+',
+        required=False,
+        help="space separated list of related scea accession(s)"
+    )
 
     args = parser.parse_args()
-    work_dir = f"script_spreadsheets/{os.path.splitext(os.path.basename(args.data))[0]}"
+    work_dir = f"script_spreadsheets/{os.path.splitext(os.path.basename(args.spreadsheet))[0]}"
+
+    tracking_sheet = utils.get_tracker_google_sheet()
+
+    if args.accession_number:
+        accession_number = args.accession_number
+    else:
+        accessions = list(tracking_sheet['data_accession']) + list(tracking_sheet['scea_accession'])
+        accessions_uniq = utils.get_unique_accessions(accessions)
+        echad_accessions = utils.get_echad_accessions(accessions_uniq)
+        accession_number = utils.get_next_echad_accession(echad_accessions)
 
-    project_info = {"accession": args.accessionnumber, "curators": args.curators}
-    spreadsheets = extract_csv_from_spreadsheet(work_dir, args.data)
-    project_details = prepare_protocol_map(work_dir, spreadsheets, project_info)
+    project_info = {"accession": accession_number, "curators": args.curators}
+    spreadsheets = extract_csv_from_spreadsheet(work_dir, args.spreadsheet)
+    project_details = prepare_protocol_map(work_dir, spreadsheets, project_info, tracking_sheet, args)
     # Save file
     with open(f"{work_dir}/project_details.json", "w") as project_details_file:
         json.dump(project_details, project_details_file, indent=2)
 
     #fpath = input("Enter file path for updated project details file: ")
-    fpath = '/Users/ami/HCA/hca-to-scea-tools/hca2scea-backend/spreadsheets/DevelopingCardiacSystem_test/project_details.json'
+    fpath = f"{work_dir}/updated_project_details.json"
```

Comment:
It might be better to keep fpath as user input instead of hard coding it. The script might crash if the user hasn't placed the json file in the correct directory. It also forces the user to places the json file in the work dir.

Reply:
Yes, agree. For testing during development I find it easier having it hardcoded, but I should haven't pushed this from my local directory. Will change it back.